src/core/reference/src/op/convert.cpp

```diff
@@ -45,16 +45,6 @@ void jit_convert_vec<float16, float>(jit::Generator& gen, const Xbyak::RegExp& s
     gen.vmovups(gen.yword[dst], f32vec);
 }
 
-template <>
-void jit_convert_vec<float, float16>(jit::Generator& gen, const Xbyak::RegExp& src, const Xbyak::RegExp& dst) {
-    auto f16vec = gen.xmm3;
-    auto f32vec = gen.ymm4;
-
-    gen.vmovups(f32vec, gen.yword[src]);
-    gen.vcvtps2ph(f16vec, f32vec, 0);
```

Comment:
Hi, @pavel-esir : could you please have a try keep this 'jit_convert_vec<float, float16>()' function while only change 'gen.vcvtps2ph(f16vec, f32vec, 0);' to 'gen.vcvtps2ph(f16vec, f32vec, 0x4);', does this change could fix the issue that you are facing?

BTW, is there any testcase to reproduce this issue?

Reply:
We were always doing clamping when f32 are converted to f16 to avoid infinities in f16

https://github.com/openvinotoolkit/openvino/blob/e7c60bb18a6f0e524890cbacc71adc7bb3837b05/src/common/transformations/src/transformations/convert_precision.cpp#L1082

For some reason evaluate for Convert operation was left with no clamping.  Unfortunately gen.vcvtps2ph(f16vec, f32vec, **0x4**) will not fix inf overflows.

Test case, i will add case reproducing this issue. Thanks for noticing this.

Reply:
The Convert op is doing conversion without clamping. Add clamping to some types will add some inconsistency as some types will be clamped some not.
Why not add Clamp operator for required precisions?
If there is some overflow in constant folding and calculated values are not valid dimension values it can mean that data are not correct.

Reply:
I added unit-tests to `tests/utils/convert_precision.cpp` which reproduce the problem being solved in this PR

https://github.com/openvinotoolkit/openvino/pull/24614/files#diff-925f6a7b0d71dbf9eba0d6929b0ab1b5d5ca190a5bb0479729e27bf60637d506